src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,67 @@
 package core.basesyntax;
 
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.List;
+
 public class WorkWithFile {
+    private static final String SUPPLY_OPERATION = "supply";
+    private static final String BUY_OPERATION = "buy";
+    private static final String RESULT_OPERATION = "result";
+    private static final String COMMA = ",";
+
+    private static final int OPERATION_TYPE = 0;
+    private static final int AMOUNT = 1;
+
     public void getStatistic(String fromFileName, String toFileName) {
+        List<String> linesFromFile = readFromFile(fromFileName);
+        String report = createReport(linesFromFile);
+        writeToFile(toFileName, report);
+    }
+
+    public String createReport(List<String> linesFromFile) {
+        String[] values;
+        StringBuilder stringBuilder = new StringBuilder();
+        int supplyValue = 0;
+        int buyValue = 0;
+        int resultValue = 0;
+        for (String str : linesFromFile) {
+            values = str.split(",");
+            try {
+                if (values[OPERATION_TYPE].equals(SUPPLY_OPERATION)) {
+                    supplyValue += Integer.parseInt(values[AMOUNT]);
+                } else if (values[OPERATION_TYPE].equals(BUY_OPERATION)) {
+                    buyValue += Integer.parseInt(values[AMOUNT]);
+                }
+            } catch (NumberFormatException e) {
+                throw new RuntimeException("Failed to convert value to number " + e);
+            }
+        }
+        resultValue = supplyValue - buyValue;
+        stringBuilder.append(SUPPLY_OPERATION + COMMA + supplyValue + System.lineSeparator())
```

Comment:
there is no sense to use StringBuilder and concatenate strings inside. eash time you are doing + you create new String